### PR TITLE
[https://nvbugs/6082303][fix] Treat <tool_call> as implicit end-of-reasoning in nano-v3 parser

### PR DESCRIPTION
**Pull request by tijyojwad**
When serving Nemotron-3-Super with --reasoning_parser nano-v3, the model occasionally omits </think> before generating <tool_call>. The parser stayed in reasoning mode and absorbed the tool call markup into reasoning_content, causing the downstream tool parser to never see it (~2-8% of streaming tool-call requests silently failed).

Treat <tool_call> as an implicit </think> in NemotronV3ReasoningParser, following the same pattern as KimiK2ReasoningParser which treats <|tool_calls_section_begin|> as an implicit reasoning end.

Made-with: Cursor

## Summary by CodeRabbit

* **New Features**
  * Reasoning parser now recognizes `<tool_call>` tags as an implicit end-of-reasoning boundary, improving handling of tool invocations within reasoning blocks in both streaming and non-streaming modes.

* **Tests**
  * Extended test coverage to validate tool call tag handling across various scenarios, including streaming and edge cases.

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: tensorrt_llm/llmapi/reasoning_parser.py [hunk 1]
```diff
@@ -318,10 +334,29 @@ def _maybe_swap_content(
         return result
 
     def parse_delta(self, delta_text: str) -> ReasoningParserResult:
-        """Wraps the parent parse_delta to track accumulated reasoning when
-        force_nonempty_content is set. When the closing tag is found
-        (in_reasoning transitions from True to False), the accumulation
-        is cleared to free memory."""
+        """Wraps the parent parse_delta to also treat ``<tool_call>`` as an
```
Comment (2ez4bz) [inline tensorrt_llm/llmapi/reasoning_parser.py]: Nit: `sed -i 's/``/`/g'`.

File: tests/unittest/llmapi/test_reasoning_parser.py [hunk 2]
```diff
@@ -326,31 +438,42 @@ def test_nano_v3_reasoning_parser_stream(delta_texts: list, content: list,
     for i, delta_text in enumerate(delta_texts):
         result = reasoning_parser.parse_delta(delta_text)
         print(f"delta_text: {delta_text}, result: {result}")
-        assert result.content == content[i]
-        assert result.reasoning_content == reasoning_context[i]
+        assert result.content == content[i], \
+            f"Step {i}: delta={delta_text!r}, expected content={content[i]!r}, got {result.content!r}"
```
Comment (2ez4bz) [inline tests/unittest/llmapi/test_reasoning_parser.py]: Nit: the debug strings are unnecessary as `pytest` will print them anyway upon failure (it overrides `assert`).

File: tensorrt_llm/llmapi/reasoning_parser.py [hunk 3]
```diff
@@ -318,10 +334,29 @@ def _maybe_swap_content(
...
+
+        ``<tool_call>`` is a special token that always arrives as a single
+        atomic delta, so we only need to check ``delta_text`` (not the
+        parent's internal buffer)."""
+        if (self.in_reasoning and self._tool_call_start in delta_text
+                and self.reasoning_end not in self._buffer):
+            remaining = self._buffer
+            self._buffer = ""
+            self.in_reasoning = False
+            tool_idx = delta_text.find(self._tool_call_start)
```
Comment (2ez4bz) [inline tensorrt_llm/llmapi/reasoning_parser.py:352]: Nit: maybe leave a comment that this is for sure non-negative, since we checked the existence of `self._tool_call_start` at line 346?

**Review by 2ez4bz — APPROVED** [review]

**Review by nv-guomingz — APPROVED** [review]
